sad_comparison_functions.py

```diff
@@ -48,6 +49,23 @@ def get_pred_iterative(cdf_obs, dist, *pars):
                 return np.array(abundance)
         i += 1
 
+def get_sample_multi_dists(S, dist_name, pars):
+    """Returns a random sample of length S from the designated distribution."""
+    dist = 0
+    if dist_name == 'logser': dist = sd.logser
+    elif dist_name == 'geom': dist = sd.geom
+    elif dist_name == 'zipf': dist = sd.zipf
+    if dist:
+        rand_smp = dist.rvs(pars, size = S)
+    else:
+        if dist_name == 'negbin':
+            n, p = pars
+            rand_smp = md.nbinom_lower_trunc.rvs(n, p, size = S)
+        elif dist_name == 'pln':
+            mu, sigma = pars
+            rand_smp = md.pln.rvs(mu, sigma, True, size = S)
+    return rand_smp
+
```

Comment:
You could also use a dictionary here to make things a little more readable. E.g. (untested code):
```python
def get_sample_multi_dists(S, dist_name, pars):
    """Returns a random sample of length S from the designated distribution."""
    rvs = {'logser': sd.logser.rvs(pars, size=S),
           'geom': sd.geom.rvs(pars, size=S),
           'zipf': sd.zipf.rvs(pars, size=S),
           'negbin': md.nbinom_lower_trunc.rvs(pars*, size=S),
           'pln': md.pln.rvs(pars*, True, size=S)}
    rand_smp = rvs[dist_name]
```